x/auction/keeper/abci.go

```diff
@@ -0,0 +1,54 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/peggyjv/sommelier/v4/x/auction/types"
+)
+
+// BeginBlocker is called at the beginning of every block
+func (k Keeper) BeginBlocker(ctx sdk.Context) {}
+
+// EndBlocker is called at the end of every block
+func (k Keeper) EndBlocker(ctx sdk.Context) {
+	// Auction price updates
+	for _, auction := range k.GetActiveAuctions(ctx) {
+		if ((ctx.BlockHeight() - int64(auction.StartBlock)) % int64(auction.PriceDecreaseBlockInterval)) == 0 {
+			// TODO post MVP (pbal) Make a more intricate & responsive step function for auction price updates
+
+			// Simple constant decrease rate meant for MVP
+			priceDecreaseAmountInUsomm := auction.InitialUnitPriceInUsomm.Mul(auction.CurrentPriceDecreaseRate)
+			newUnitPriceInUsomm := auction.CurrentUnitPriceInUsomm.Sub(priceDecreaseAmountInUsomm)
+
+			// If the new price would be non positive, finish the auction
+			if newUnitPriceInUsomm.LTE(sdk.NewDec(0)) {
+				err := k.FinishAuction(ctx, auction)
```

Comment:
This error seems like it would only be populated if there is a failure sending tokens from module to module. Should we halt the chain if that fails?

Reply:
@EricBolten & I discussed this previously and I think we arrived on the idea that if a module to module token transfer fails, something is fundamentally wrong and we should panic (as we do on the next couple lines below).